homeassistant/components/influxdb.py

```diff
@@ -77,6 +77,11 @@ def influx_event_listener(event):
             _state = 0
         else:
             _state = state.state
+            if _state and isinstance(_state, str):
```

Comment:
Okay, let's use the follow then:
```python
try:
    _state = float(_state)
except ValueError:
    pass
```

Using a literal eval is a security risk which I don't want to include.